File: android/src/main/java/cn/ooix/capacitor/safearea/SafeAreaPlugin.java
package cn.ooix.capacitor.safearea;

import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "SafeArea")
public class SafeAreaPlugin extends Plugin {

    private SafeArea implementation;

    @Override
    public void load() {
        implementation = new SafeArea(getActivity());
    }

    @PluginMethod
    public void getSafeArea(PluginCall call) {
        call.resolve(implementation.getSafeArea());
    }

    @PluginMethod
    public void getStatusBarHeight(PluginCall call) {
        call.resolve(implementation.getStatusBarHeight());
    }
}

File: ios/Plugin/SafeAreaPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SafeAreaPlugin)
public class SafeAreaPlugin: CAPPlugin {
    private let implementation = SafeArea()

    @objc func getSafeArea(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            call.resolve(self.implementation.getSafeArea())
        }
    }

    @objc func getStatusBarHeight(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            call.resolve(self.implementation.getStatusBarHeight())
        }
    }
}
